[BUG] - SMB Isn't Scoped in the Code

## Description

The logs below show that `Smb`-related types aren't in scope. I'm going to guess this is wrong because there are clearly working builds?

## Steps to reproduce

Both default and `--no-default-features` don't work.

## Expected behaviour

Termscp should install correctly from Cargo.

## Environment

- OS: Windows 11 22H2 (`22621.1848`)
- Architecture x86_64
- Rust version: `rustc 1.70.0 (90c541806 2023-05-31)`
- termscp version: `0.12.1`

## Log

```
$ cargo install --locked --no-default-features termscp
[...]
error[E0412]: cannot find type `SmbParams` in this scope
   --> C:\Users\TheBrenny\.cargo\registry\src\index.crates.io-6f17d22bba15001f\termscp-0.12.1\src\filetransfer\builder.rs:144:27
    |
144 |     fn smb_client(params: SmbParams) -> SmbFs {
    |                           ^^^^^^^^^ not found in this scope
    |
help: consider importing one of these items
    |
5   + use crate::config::bookmarks::SmbParams;
    |
5   + use crate::filetransfer::params::SmbParams;
    |

error[E0412]: cannot find type `SmbFs` in this scope
   --> C:\Users\TheBrenny\.cargo\registry\src\index.crates.io-6f17d22bba15001f\termscp-0.12.1\src\filetransfer\builder.rs:144:41
    |
144 |     fn smb_client(params: SmbParams) -> SmbFs {
    |                                         ^^^^^ not found in this scope

error[E0433]: failed to resolve: use of undeclared type `SmbParams`
   --> C:\Users\TheBrenny\.cargo\registry\src\index.crates.io-6f17d22bba15001f\termscp-0.12.1\src\utils\parser.rs:316:37
    |
316 |                 ProtocolParams::Smb(SmbParams::new(address, share).username(username)),
    |                                     ^^^^^^^^^ use of undeclared type `SmbParams`
    |
help: consider importing one of these items
    |
6   + use crate::config::bookmarks::SmbParams;
    |
6   + use crate::filetransfer::params::SmbParams;
    |

error[E0433]: failed to resolve: use of undeclared type `SmbCredentials`
   --> C:\Users\TheBrenny\.cargo\registry\src\index.crates.io-6f17d22bba15001f\termscp-0.12.1\src\filetransfer\builder.rs:145:31
    |
145 |         let mut credentials = SmbCredentials::new(params.address, params.share);
    |                               ^^^^^^^^^^^^^^ use of undeclared type `SmbCredentials`

error[E0433]: failed to resolve: use of undeclared type `SmbFs`
   --> C:\Users\TheBrenny\.cargo\registry\src\index.crates.io-6f17d22bba15001f\termscp-0.12.1\src\filetransfer\builder.rs:154:9
    |
154 |         SmbFs::new(credentials)
    |         ^^^^^ use of undeclared type `SmbFs`

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `termscp` (bin "termscp") due to 5 previous errors
error: failed to compile `termscp v0.12.1`, intermediate artifacts can be found at `C:\Users\THEBRE~1\AppData\Local\Temp\cargo-installrtJ2KG`
```

--- a/build.rs
+++ b/build.rs
@@ -10,6 +10,7 @@ fn main() {
         windows: { target_family = "windows" },
         // exclusive features
         smb: { all(feature = "smb", not( macos )) },
-        smb_unix: { all(unix, feature = "smb", not(macos)) }
+        smb_unix: { all(unix, feature = "smb", not(macos)) },
+        smb_windows: { all(windows, feature = "smb") }
     }
 }

--- a/src/filetransfer/builder.rs
+++ b/src/filetransfer/builder.rs
@@ -140,7 +140,7 @@ impl Builder {
         }
     }
 
-    #[cfg(windows)]
+    #[cfg(smb_windows)]
     fn smb_client(params: SmbParams) -> SmbFs {
         let mut credentials = SmbCredentials::new(params.address, params.share);
 

--- a/src/ui/activities/setup/actions.rs
+++ b/src/ui/activities/setup/actions.rs
@@ -366,7 +366,7 @@ impl SetupActivity {
             .get_color(&Id::Theme(IdTheme::StatusSync))
             .map_err(|_| Id::Theme(IdTheme::StatusSync))?;
         // Update theme
-        let mut theme: &mut Theme = self.theme_mut();
+        let theme = self.theme_mut();
         theme.auth_address = auth_address;
         theme.auth_bookmarks = auth_bookmarks;
         theme.auth_password = auth_password;

--- a/src/utils/parser.rs
+++ b/src/utils/parser.rs
@@ -73,7 +73,7 @@ static REMOTE_SMB_OPT_REGEX: Lazy<Regex> = lazy_regex!(
  * - group 3: share
  * - group 4: remote-dir?
  */
-#[cfg(windows)]
+#[cfg(smb_windows)]
 static REMOTE_SMB_OPT_REGEX: Lazy<Regex> =
     lazy_regex!(r"(?:([^@]+)@)?(?:([^:\\]+))(?:\\([^\\]+))?(?:(\\.+))?");
 
@@ -171,7 +171,7 @@ fn parse_remote_opt_protocol(
             let protocol = match protocol {
                 Some(Ok(protocol)) => protocol,
                 Some(Err(err)) => return Err(err),
-                #[cfg(windows)]
+                #[cfg(smb_windows)]
                 None if groups.get(2).is_some() => FileTransferProtocol::Smb,
                 None => default,
             };
@@ -295,7 +295,7 @@ fn parse_smb_remote_opts(s: &str) -> Result<FileTransferParams, String> {
     }
 }
 
-#[cfg(windows)]
+#[cfg(smb_windows)]
 fn parse_smb_remote_opts(s: &str) -> Result<FileTransferParams, String> {
     match REMOTE_SMB_OPT_REGEX.captures(s) {
         Some(groups) => {
@@ -639,7 +639,7 @@ mod tests {
     }
 
     #[test]
-    #[cfg(windows)]
+    #[cfg(smb_windows)]
     fn should_parse_smb_address() {
         let result = parse_remote_opt(&String::from("\\\\myserver\\myshare"))
             .ok()
@@ -652,7 +652,7 @@ mod tests {
     }
 
     #[test]
-    #[cfg(windows)]
+    #[cfg(smb_windows)]
     fn should_parse_smb_address_with_opts() {
         let result = parse_remote_opt(&String::from("\\\\omar@myserver\\myshare\\path"))
             .ok()

